src/main/java/jenkins/plugins/logstash/persistence/BuildData.java

```diff
@@ -212,21 +212,14 @@ public BuildData(AbstractBuild<?, ?> build, Date currentTime, TaskListener liste
   }
 
   // Pipeline project build
-  public BuildData(Run<?, ?> build, Date currentTime, TaskListener listener) {
+  public BuildData(Run<?, ?> build, Date currentTime, TaskListener listener, hudson.EnvVars envVars) {
     initData(build, currentTime);
 
+    buildVariables = envVars;
     rootProjectName = projectName;
     rootFullProjectName = fullProjectName;
     rootProjectDisplayName = displayName;
     rootBuildNum = buildNum;
-
-    try {
-      // TODO: sensitive variables are not filtered, c.f. https://stackoverflow.com/questions/30916085
-      buildVariables = build.getEnvironment(listener);
-    } catch (IOException | InterruptedException e) {
-      LOGGER.log(WARNING,"Unable to get environment for " + build.getDisplayName(),e);
-      buildVariables = new HashMap<>();
-    }
```

Comment:
I don't catch. I thought that this part of code (line 215. public BuildData(Run<?, ?> build, Date currentTime, TaskListener listener, hudson.EnvVars envVars) {) was on "ly for declarative pipeline and not for non pipeline builds.
That's why I've considered that "envars" could not be null and that it was not necessary to do the build.getEnvironment(listener)
Note that my concern is to get more info than build.getEnvironment(listener) because with this command, you will not recover the environment variables of the stage itself and you will not get the STAGE_NAME and NODE_NAME that are to my mind very important variables that needs to be pushed into a Graylog. Otherwise you cannot make any correlation between info that might be recovered of one line of log and the stage and node that was at the origin of this line of log.

Reply:
I think best way to resolve this would be to add some automated tests to show it's working

Reply:
> I'm missing a test that ensures that the information you want is included.
> Another topic. What envvars are injected by the context? Do they contain sensitive information maybe like passwords?

You are probably right, but that need to be checked. 
I've no idea how to distinguish sensitive information in a set of environment variables.

Reply:
Filtering for sensitive variables is not supported in Pipeline. At least it wasn't last time I checked. 
I think it's good enough if we make sure the sensitive variables for old-style builds are filtered

Reply:
This constructor is indeed not called for instances of AbstractBuild. But it is not limited to WorkflowRun. It will also be called for any other implementation directly inheriting from Run (e.g. AsyncRun, ExternalRun, see https://jenkins.io/doc/developer/extensions/jenkins-core/#run)
So I would prefer to keep this code.
(Though any of these job types would never have been working together with logstash plugin before the pipeline support was added)

Reply:
OK. I suggest to keep this code but to execute it only if enVars is non null or empty if I figure out to create a new contructor (I guess it's related with your comment src/main/java/jenkins/plugins/logstash/LogstashNotifier.java  line 110).